nimble/host/services/cts/src/ble_svc_cts.c

```diff
@@ -0,0 +1,544 @@
+static uint16_t ble_svc_cts_current_time_handle;
+#endif
+
+/* Connection handle 
+ *
+ * TODO: In order to support multiple connections we would need to save
+ *       the handles for every connection, not just the most recent. Then
+ *       we would need to notify each connection when needed.
+ * */
+static uint16_t ble_svc_cts_conn_handle;
+
+/* Access function */
+static int
+ble_svc_cts_access(uint16_t conn_handle, uint16_t attr_handle,
+                   struct ble_gatt_access_ctxt *ctxt, void *arg);
+
+static const struct ble_gatt_svc_def ble_svc_cts_defs[] = {
+    {
+        /*** Current Time Service. */
+        .type = BLE_GATT_SVC_TYPE_PRIMARY,
+        .uuid = BLE_UUID16_DECLARE(BLE_SVC_CTS_UUID16),
+        .characteristics = (struct ble_gatt_chr_def[]) { {
+	    /*** Current time characteristic */
+            .uuid = BLE_UUID16_DECLARE(BLE_SVC_CTS_CHR_UUID16_CURRENT_TIME),
+            .access_cb = ble_svc_cts_access,
+#if MYNEWT_VAL(BLE_SVC_CTS_CURRENT_TIME_NOTIFY_ENABLE) > 0
+	    .val_handle = &ble_svc_cts_current_time_handle,
+#endif
+            .flags = BLE_GATT_CHR_F_READ |
+	             MYNEWT_VAL(BLE_SVC_CTS_CURRENT_TIME_READ_PERM) |
+#if MYNEWT_VAL(BLE_SVC_CTS_CURRENT_TIME_NOTIFY_ENABLE) > 0
+	             BLE_GATT_CHR_F_NOTIFY |
+#endif
+#if MYNEWT_VAL(BLE_SVC_GAP_DEVICE_NAME_WRITE_PERM) >= 0
+                     BLE_GATT_CHR_F_WRITE |
+                     MYNEWT_VAL(BLE_SVC_CTS_CURRENT_TIME_WRITE_PERM) |
+#endif
+	             0,
+	    }, {
+#if MYNEWT_VAL(BLE_SVC_CTS_LOCAL_TIME_INFORMATION_READ_PERM) >= 0
+	    /*** Local time information characteristic */
+            .uuid = BLE_UUID16_DECLARE(BLE_SVC_CTS_CHR_UUID16_LOCAL_TIME_INFORMATION),
+            .access_cb = ble_svc_cts_access,
+            .flags = BLE_GATT_CHR_F_READ |
+	             MYNEWT_VAL(BLE_SVC_CTS_LOCAL_TIME_INFORMATION_READ_PERM) |
+#if MYNEWT_VAL(BLE_SVC_CTS_LOCAL_TIME_INFORMATION_WRITE_PERM) >= 0
+                     BLE_GATT_CHR_F_WRITE |
+                     MYNEWT_VAL(BLE_SVC_CTS_LOCAL_TIME_INFORMATION_WRITE_PERM) |
+#endif
+	             0,
+	    }, {
+#endif
+#if MYNEWT_VAL(BLE_SVC_CTS_REFERENCE_TIME_INFORMATION_READ_PERM) >= 0
+	    /*** Reference time information characteristic */
+            .uuid = BLE_UUID16_DECLARE(BLE_SVC_CTS_CHR_UUID16_REFERENCE_TIME_INFORMATION),
+            .access_cb = ble_svc_cts_access,
+            .flags = BLE_GATT_CHR_F_READ |
+	             MYNEWT_VAL(BLE_SVC_CTS_REFERENCE_TIME_INFORMATION_READ_PERM),
+	    }, {
+#endif
+            0, /* No more characteristics in this service. */
+        } },
+    },
+
+    {
+        0, /* No more services. */
+    },
+};
+
+static inline int
+ble_svc_cts_current_time_read_access(struct ble_gatt_access_ctxt *ctxt)
+{
+    /* Get current clock time */
+    struct os_timeval  tv;
+    struct os_timezone tz;
+    struct clocktime   ct;
+    
+    if (os_gettimeofday(&tv, &tz) || timeval_to_clocktime(&tv, &tz, &ct)) {
```

Comment:
this introduces dependency to yet another Mynewt compoments thus making this service only usable with Mynewt - we would prefer something more generic